imod/mf6/wel.py

```diff
@@ -410,242 +688,280 @@ def __create_assigned_wells(
-        See also: https://water.usgs.gov/water-resources/software/MODFLOW-6/mf6io_6.4.0.pdf#page=35
+        if "filt_top" not in df.columns or "filt_bot" not in df.columns:
+            log_msg = textwrap.dedent(
+                f"""
+                In well {key} the filt_top and filt_bot columns were not both found;
+                this is not supported for import."""
+            )
+            logger.log(loglevel=LogLevel.ERROR, message=log_msg, additional_depth=2)
+            raise ValueError(log_msg)
 
-        Note
-        ----
-        The "layer" coordinate should already be provided in the dataframe.
-        To determine the layer coordinate based on screen depts, look at
-        :func:`imod.prepare.wells.assign_wells`.
+        # Groupby unique wells, to get dataframes per time.
+        colnames_group = ["x", "y", "filt_top", "filt_bot", "id"]
+        wel_index, unique_well_groups = zip(*df.groupby(colnames_group))
 
-        Parameters
-        ----------
-        dst_grid: {xr.DataArray, xu.UgridDataArray}
-            Destination grid to map the points to based on their x and y coordinates.
-        x: {list, np.array}
-            array-like with x-coordinates
-        y: {list, np.array}
-            array-like with y-coordinates
-        layer: {list, np.array}
-            array-like with layer-coordinates
+        # Unpack wel indices by zipping
+        x, y, filt_top, filt_bot, id = zip(*wel_index)
+        well_rate = _prepare_well_rates_from_groups(unique_well_groups, times)
 
-        Returns
-        -------
-        cellid : xr.DataArray
-            2D DataArray with a ``ncellid`` rows and 3 to 2 columns, depending
-            on whether on a structured or unstructured grid."""
+        return cls(
+            x=np.array(x, dtype=float),
+            y=np.array(y, dtype=float),
+            screen_top=np.array(filt_top, dtype=float),
+            screen_bottom=np.array(filt_bot, dtype=float),
+            rate=well_rate,
+            minimum_k=minimum_k,
+            minimum_thickness=minimum_thickness,
+        )
 
-        # Find indices belonging to x, y coordinates
-        indices_cell2d = points_indices(dst_grid, out_of_bounds="ignore", x=x, y=y)
-        # Convert cell2d indices from 0-based to 1-based.
-        indices_cell2d = {dim: index + 1 for dim, index in indices_cell2d.items()}
-        # Prepare layer indices, for later concatenation
 
-        if isinstance(dst_grid, xu.UgridDataArray):
-            indices_layer = xr.DataArray(
-                layer, coords=indices_cell2d["mesh2d_nFaces"].coords
-            )
-            face_dim = dst_grid.ugrid.grid.face_dimension
-            indices_cell2d_dims = [face_dim]
-            cell2d_coords = ["cell2d"]
-        else:
-            indices_layer = xr.DataArray(layer, coords=indices_cell2d["x"].coords)
-            indices_cell2d_dims = ["y", "x"]
-            cell2d_coords = ["row", "column"]
+class LayeredWell(GridAgnosticWell):
```

Comment:
please add an instance of this class to the package sanity check tests

Reply:
Done